version: '3.1'
services:
  vault:
    image: vault:latest
    ports:
      - "8200:8200"
    volumes:
      - ./vault:/vault/
    cap_add:
      - IPC_LOCK
    entrypoint: vault server -config=vault/vault.json

  db:
    image: postgres:14.5
    ports:
      - 5432:5432
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -d db" ]
      interval: 10s
      timeout: 5s
      retries: 5
    env_file:
      - docker.env

  init_db:
    depends_on:
      - db
      - vault
    build: .
    command: bash -c "python3 src/00-dataset_to_db.py"
    image: 54367888/mle-template:latest
    volumes:
      - ./vault:/app/vault/
    env_file:
      - docker.env
    links:
      - "db:database"
      - "vault:vault"

  fit_predict:
    depends_on:
      - init_db
    build: .
    command: bash -c "python3 src/bikes_regression.py"
    image: 54367888/mle-template:latest
    volumes:
      - ./vault:/app/vault/
    env_file:
      - docker.env
    links:
      - "db:database"
      - "vault:vault"

  tests:
    build: .
    command: bash -c "python3 -m pytest tests/test_model_quality.py"
